File: problems/66. Plus One/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if not digits:
            return [1]
        i = len(digits) - 1
        digits[i] += 1

        while i > 0 and digits[i] > 9:
            digits[i] = 0
            digits[i-1] += 1
            i -= 1

        if digits[i] > 9:
            digits[i] = 0
            digits.append(1)
            digits[i], digits[-1] = digits[-1], digits[i]

        return digits

File: problems/66. Plus One/solution.cpp
class Solution
{
public:
    vector<int> plusOne(vector<int> &digits)
    {
        if (!digits.size())
            return digits;
        int i = digits.size() - 1, carry = 1;

        while (i >= 0)
        {
            if (!carry)
                break;
            digits[i] += carry;
            carry = 0;
            if (digits[i] == 10)
            {
                digits[i] = 0;
                if (i == 0)
                    digits.insert(digits.begin(), 1);
                carry++;
            }
            i--;
        }

        return digits;
    }
};
